homeassistant/components/statistics/sensor.py

```diff
@@ -472,8 +478,8 @@ async def async_update(self) -> None:
         if timestamp := self._next_to_purge_timestamp():
             _LOGGER.debug("%s: scheduling update at %s", self.entity_id, timestamp)
             if self._update_listener:
-                self._update_listener()
-                self._update_listener = None
+                self._update_listener()  # pragma: no cover
+                self._update_listener = None  # pragma: no cover
```

Comment:
When does this happen?

Reply:
Can't exactly tell you. I did not develop this part of the code. Pytest always showed missing coverage for these two lines, hence the comment. Maybe the code can be removed or changed, but I didn't feel the need.

Reply:
There shouldn't be any `pragma: no cover` in homeassistant code.
It is better to have reduced coverage than invalid coverage.

Reply:
Opened #88173

Reply:
If the code is there, there must be a reason...

- If it is dead code that can't be reached, then it should be removed
- If it is not dead code, then we need to be reminded that coverage should be improved

The only time when `no cover` is in use is when we raise an exception for unreachable code.

Reply:
There is not reason to introduce fake coverage. Epenet is right here, it should not have had these flags. I've therefore merged the PR to remove them.

Reply:
On second thought @epenet I agree with you. I did not develop this code nor did I ever touch it. I believe it's "dead code" as you called it and will do some tests and provide a fix in a future PR.